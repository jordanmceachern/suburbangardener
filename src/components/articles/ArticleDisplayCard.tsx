"use client";

import React from "react";
import Link from "next/link";
import BackgroundImage from "../BackgroundImage";

export interface ArticleDisplayData {
  id: string;
  title: string;
  excerpt: string;
  publishedDate: string;
  author: string;
  slug: string;
  imageUrl: string;
  category: string;
  content?: string; // Full article content for individual pages
}

interface ArticleDisplayCardProps {
  article: ArticleDisplayData;
  className?: string;
  onReadMore?: () => void;
  buttonVariant?: "default" | "narrow";
  showArrow?: boolean;
  titleClassName?: string;
  imageClassName?: string;
}

export default function ArticleDisplayCard({
  article,
  className = "",
  onReadMore,
  buttonVariant = "default",
  showArrow = false,
  titleClassName = "",
  imageClassName = "",
}: ArticleDisplayCardProps) {
  const handleReadMore = () => {
    console.log("Read More clicked for article:", article.slug);
    if (onReadMore) {
      onReadMore();
    }
  };

  return (
    <article
      className={`bg-primary-50 dark:bg-slate-500 rounded-lg shadow-lg overflow-hidden min-w-0 flex flex-col ${className}`}
    >
      <div className={`h-48 sm:h-64 bg-slate-800 ${imageClassName}`}>
        <BackgroundImage
          src={article.imageUrl}
          label={article.title}
          className="w-full h-full rounded-b-none"
        />
      </div>
      <div className="px-6 pb-6 py-4 flex flex-col flex-grow">
        <div className="flex items-center mb-2">
          <span className="text-neutral-700 dark:text-neutral-200 text-sm">
            {article.publishedDate}
          </span>
        </div>
        <h2
          className={`text-xl lg:text-2xl xl:text-3xl font-bold text-neutral-900 dark:text-neutral-100 mb-3 ${titleClassName}`}
        >
          {article.title}
        </h2>
        <p className="text-neutral-700 dark:text-neutral-200 text-base sm:text-lg mb-4 sm:mb-6 line-clamp-3">
          {article.excerpt}
        </p>

        {onReadMore ? (
          <button
            onClick={handleReadMore}
            className={`bg-tertiary-500 hover:bg-tertiary-600 dark:bg-tertiary-600 dark:hover:bg-tertiary-700 text-white px-4 py-2 sm:px-6 sm:py-3 rounded-lg font-medium transition-colors mt-auto ${
              buttonVariant === "narrow" ? "self-center" : ""
            }`}
          >
            Read More{showArrow ? " →" : ""}
          </button>
        ) : (
          <Link
            href={`/articles/${article.slug}`}
            className={`bg-tertiary-500 hover:bg-tertiary-600 dark:bg-tertiary-600 dark:hover:bg-tertiary-700 text-white px-4 py-2 sm:px-6 sm:py-3 rounded-lg font-medium transition-colors mt-auto inline-block text-center ${
              buttonVariant === "narrow" ? "self-center" : ""
            }`}
          >
            Read More{showArrow ? " →" : ""}
          </Link>
        )}
      </div>
    </article>
  );
}
