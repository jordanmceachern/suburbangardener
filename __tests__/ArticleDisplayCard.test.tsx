import { render } from "@testing-library/react";
import ArticleDisplayCard from "@/components/articles/ArticleDisplayCard";

describe("ArticleDisplayCard Component", () => {
  const mockArticle = {
    id: "1",
    title: "Test Article",
    excerpt: "This is a test article excerpt",
    slug: "test-article",
    content: "Test content",
    imageUrl: "/test-image.jpg",
    author: "Test Author",
    publishedDate: "2024-01-01",
    category: "gardening",
  };

  it("should match DOM snapshot", () => {
    const { container } = render(<ArticleDisplayCard article={mockArticle} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should maintain CSS classes for card layout", () => {
    const { container } = render(<ArticleDisplayCard article={mockArticle} />);

    // Check for card-related classes
    const hasCardClasses =
      container.innerHTML.includes("card") ||
      container.innerHTML.includes("border") ||
      container.innerHTML.includes("rounded") ||
      container.innerHTML.includes("shadow") ||
      container.innerHTML.includes("bg-");

    expect(hasCardClasses).toBe(true);
  });
});
