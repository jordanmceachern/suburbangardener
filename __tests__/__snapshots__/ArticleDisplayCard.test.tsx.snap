// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ArticleDisplayCard Component should match DOM snapshot 1`] = `
<article
  class="bg-primary-50 dark:bg-slate-500 rounded-lg shadow-lg overflow-hidden min-w-0 flex flex-col "
>
  <div
    class="h-48 sm:h-64 bg-slate-800 "
  >
    <div
      class="h-auto w-full flex justify-center w-full h-full rounded-b-none"
    >
      <div
        aria-label="Test Article"
        class="bg-contain bg-center bg-no-repeat w-full h-full  rounded-lg"
        style="background-image: url("/test-image.jpg");"
      />
    </div>
  </div>
  <div
    class="px-6 pb-6 py-4 flex flex-col flex-grow"
  >
    <div
      class="flex items-center mb-2"
    >
      <span
        class="text-neutral-700 dark:text-neutral-200 text-sm"
      >
        2024-01-01
      </span>
    </div>
    <h2
      class="text-xl lg:text-2xl xl:text-3xl font-bold text-neutral-900 dark:text-neutral-100 mb-3 "
    >
      Test Article
    </h2>
    <p
      class="text-neutral-700 dark:text-neutral-200 text-base sm:text-lg mb-4 sm:mb-6 line-clamp-3"
    >
      This is a test article excerpt
    </p>
    <a
      class="bg-tertiary-500 hover:bg-tertiary-600 dark:bg-tertiary-600 dark:hover:bg-tertiary-700 text-white px-4 py-2 sm:px-6 sm:py-3 rounded-lg font-medium transition-colors mt-auto inline-block text-center "
      href="/articles/test-article"
    >
      Read More
    </a>
  </div>
</article>
`;
